### PR TITLE
creational/abstract-factory: implement abstract pattern

### PR DESCRIPTION
**Pull request by Inux**
Hi Tamer

It works now :)
The problem was that I tried to make an remote branch to compare to master branch. But I had no permission. Then I realized that I can make a fork to compare to the master branch :)

Hope it is good enough to integrate (still in the learning curve)

Best regards

Steve aka Inux

### CODE REVIEW

File: creational/abstract_factory.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	return &otherOSButton{}
+}
+
+func main() {
+	var factory iGUIFactory
+
+	switch runtime.GOOS {
+	case "windows":
+		factory = &winFactory{}
+		break
```
Comment (bndw) [inline creational/abstract_factory.go]: No need to `break` in Go switch statements

Comment (Inux) [reply creational/abstract_factory.go]: I've made a new commit. Hope it is fine now

**Comment by Inux** [comment]
@bndw is there still anything wrong on this PR? Just wondering why it is not already integrated

**Comment by LinMAD** [comment]
Hi Inux.

MB you forgot to update README.md mark status form "x" to "✔" for Creational Patterns -> Abstract Factory


Cheers

**Comment by Inux** [comment]
Hi @LinMAD,

I tried to add the ✔ but somehow there is a conflict appearing. Looks like I have no permissions to change the README

Cheers

**Comment by LinMAD** [comment]
Hi @Inux .

I think it's can be related with margin lates master to your branch, so MB, try to rebase your branch with the latest master branch, for example. ```git rebase upstream/master```

Before do, make a local backup branch, rebase can be evil :) And also make sure your fork contains in config upstream path.

Hope my advice help ya, cheers.

**Pull request closed without merging**